File: client/src/app/models/order.ts
export interface OrderItem {
    productId: number;
    name: string;
    image: string;
    price: number;
    quantity: number;
}

export interface Order {
    id: number;
    customerId: string;
    orderStatus: string;
    orderDate: string;
    total: number;
    paymentMethod: string;
    orderItems: OrderItem[];
}

File: API/Entities/Order/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Entities.Order
{
    public class Order
    {
        public int Id { get; set; }
        public string CustomerId { get; set; }
        public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;
        public DateTime OrderDate { get; set; } = DateTime.Now;
        public long Total { get; set; }
        public PaymentMethod PaymentMethod { get; set; }
        public List<OrderItem> OrderItems { get; set; }
    }
}